### ISSUE TITLE
Analog SDK failed to initialise: IncompatibleVersion

### ISSUE THREAD

**Issue by tonytbui** [WootingKb/wooting-analog-sdk#7]
Hi, I followed the Visual Code Studio guide and got the following error when running the example in the Initialisation step:

`Hello Analog SDK!
Unable to load library: libwooting_analog_sdk.dylib
Err: dlopen(libwooting_analog_sdk.dylib, 2): image not found
Cannot access Wooting Analog SDK function as this wrapper is for SDK major version 0, whereas the SDK has major version -1994
Analog SDK failed to initialise: IncompatibleVersion

Press any key to continue...`

**Comment by simon-wh** [comment]
It seems your Analog SDK install isn't working properly, most likely it's because the install instructions are a bit outdated (and slightly broken), could you follow [these instructions](https://github.com/WootingKb/wooting-analog-sdk/tree/develop#mac) and try again? Forgot to merge that to master.

The `Cannot access Wooting Analog SDK function as this wrapper is for SDK major version 0, whereas the SDK has major version -1994` is an issue with the wrapper that I need to resolve. As it indicates that the version is wrong when in fact it has failed to load the SDK

**Comment by tonytbui** [comment]
> It seems your Analog SDK install isn't working properly, most likely it's because the install instructions are a bit outdated (and broken), could you follow [these instructions](https://github.com/WootingKb/wooting-analog-sdk/tree/develop#mac) and try again? Forgot to merge that to master.
> 
> The `Cannot access Wooting Analog SDK function as this wrapper is for SDK major version 0, whereas the SDK has major version -1994` is an issue with the wrapper that I need to resolve. As it indicates that the version is wrong when in fact it has failed to load the SDK

Hi, the error persisted.

1. I followed instructions on getting the `wrapper/sdk/libwooting_analog_sdk.dylib` from the latest release wooting-analog-sdk-v0.4.0-x86_64-apple-darwin.tar.gz
 on [https://github.com/WootingKb/wooting-analog-sdk/releases](url). In my directory, there's now `/usr/local/lib/libwooting_analog_sdk.dylib`.
2. My ~/.bash_profile has: ```export DYLD_LIBRARY_PATH="/Library/libwooting_analog_sdk.dylib"
export PATH="/usr/local/bin:$PATH"```. 
3. `echo $PATH` shows `/Users/tonyb/opt/anaconda3/bin:/Users/tonyb/opt/anaconda3/condabin:/usr/local/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/opt/X11/bin:~/.dotnet/tools:/Library/Frameworks/Mono.framework/Versions/Current/Commands`
4. My OS is macOS Mojave 10.14.6, so I skipped the reference about security/permission.
5. From [https://github.com/WootingKb/wooting-analog-plugin/releases](url), I got the libwooting_analog_plugin.dylib file. There's now `/Library/WootingAnalogPlugins/wooting-analog-plugin/libwooting_analog_plugin.dylib` in my directory.

**Comment by tonytbui** [comment]
Quick update: I got different outputs running in Visual Studio for Mac and in terminal.

**Change made**: copied libwooting_analog_sdk.dylib into ~/Documents/analog-test/analog-test. (Adding a reference to the WootingAnalogSDK.NET NuGet package, as instructed on https://dev.wooting.io/wooting-analog-sdk-guide/c-guide/ did not automatically generate the .dylib file in the project folder.) My project directory looks like this:
```
analog-test/analog-test/Program.cs
analog-test/analog-test/analog-test.csproj
analog-test/analog-test/libwooting_analog_sdk.dylib

analog-test/analog-test/bin:
Debug

analog-test/analog-test/obj:
Debug
analog-test.csproj.nuget.g.props
analog-test.csproj.nuget.cache
analog-test.csproj.nuget.g.targets
analog-test.csproj.nuget.dgspec.json
project.assets.json
```

**Ouputs**:

1. Running on Terminal when in ~/Documents/analog-test/analog-test gave
```
Hello Analog SDK!
Analog SDK Successfully initialised with 0 devices!
```

2. Running on Terminal from elsewhere using command dotnet run --project ~/Documents/analog-test/analog-test gave
```
Hello Analog SDK!
Unable to load library: libwooting_analog_sdk.dylib
Err: dlopen(libwooting_analog_sdk.dylib, 2): image not found
Cannot access Wooting Analog SDK function as this wrapper is for SDK major version 0, whereas the SDK has major version -1994
Analog SDK failed to initialise: IncompatibleVersion
```
3. Running Program.cs in Visual Studio Community 2019 for Mac gave the same error as in 2.2.

**Comment by tonytbui** [comment]
Quick update #2: I copied the Full Code Example (from [)](https://dev.wooting.io/wooting-analog-sdk-guide/c-guide/) to the Program.cs and ran `dotnet run` while in the same containing folder. Got this:
```
Hello Analog SDK!
Analog SDK Successfully initialised with 0 devices!
Error getting devices: 0
Read failed with NoDevices
Read failed with NoDevices
Read failed with NoDevices
```

My Wooting One keyboard (S/N: A01B1809W011H01063) was connected (the background light was on, and pressing keys did work as would with a traditional keyboard).

**Comment by PastaJ36** [comment]
Have you updated the firmware with the Wootility beta client? You can find it in our [discord](https://wooting.io/discord). We're moving the new firmware to public later this week.

**Comment by tonytbui** [comment]
> Have you updated the firmware with the Wootility beta client? You can find it in our [discord](wooting.io/discord). We're moving the new firmware to public later this week.

Hi, short story: **it worked!**

Long story: the hyperlink did not work. I google-d wootility and got the latest (public) version of 3.3.3. Updated firmware through that version and did _not_ get the keyboard connected. There was a link to your discord, however, on that 3.3.3 version. I got the 3.4.1 (beta) version there and updated the firmware through that. And, it worked.
